lib/src/authentication/credential_store.dart

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: import_of_legacy_library_into_null_safe
+
+import 'dart:convert';
+
+import 'package:path/path.dart' as path;
+
+import '../io.dart';
+import '../log.dart' as log;
+import 'credential.dart';
+import 'scheme.dart';
+
+/// Stores and manages authentication credentials.
+class CredentialStore {
```

Comment:
How about we simplify all of this a bit:
 * Do away with `Credential` as an abstraction
 * Rename `CredentialStore` to `TokenStore`